From ff676f5880b2388d509e2a8e5accd952bad341f6 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Sat, 15 Apr 2023 20:24:18 +0200
Subject: [PATCH 43/46] Revert "mm: do not increment pgfault stats when page
 fault handler retries"

This reverts commit 1a5fb8ec4e3c1c0421a4ce5eaecbafbca85396da.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 mm/memory.c | 16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/mm/memory.c b/mm/memory.c
index 354229d74..0af81ac5d 100644
--- a/mm/memory.c
+++ b/mm/memory.c
@@ -5230,16 +5230,17 @@ vm_fault_t handle_mm_fault(struct vm_area_struct *vma, unsigned long address,
 
 	__set_current_state(TASK_RUNNING);
 
+	count_vm_event(PGFAULT);
+	count_memcg_event_mm(vma->vm_mm, PGFAULT);
+
 	ret = sanitize_fault_flags(vma, &flags);
 	if (ret)
-		goto out;
+		return ret;
 
 	if (!arch_vma_access_permitted(vma, flags & FAULT_FLAG_WRITE,
 					    flags & FAULT_FLAG_INSTRUCTION,
-					    flags & FAULT_FLAG_REMOTE)) {
-		ret = VM_FAULT_SIGSEGV;
-		goto out;
-	}
+					    flags & FAULT_FLAG_REMOTE))
+		return VM_FAULT_SIGSEGV;
 
 	/*
 	 * Enable the memcg OOM handling for faults triggered in user
@@ -5270,11 +5271,6 @@ vm_fault_t handle_mm_fault(struct vm_area_struct *vma, unsigned long address,
 	}
 
 	mm_account_fault(regs, address, flags, ret);
-out:
-	if (!(ret & VM_FAULT_RETRY)) {
-		count_vm_event(PGFAULT);
-		count_memcg_event_mm(vma->vm_mm, PGFAULT);
-	}
 
 	return ret;
 }
-- 
2.40.0.71.g950264636c

